extensions/brew/src/utils.ts

```diff
@@ -66,9 +66,59 @@ export async function fetchRemote<T>(remote: Remote<T>): Promise<T[]> {
 
 /// Toast
 
+interface ActionToastOptions {
+  title: string;
+  message?: string;
+  cancelable: boolean;
+}
+
+export function showActionToast(actionOptions: ActionToastOptions): AbortController | undefined {
+  const options: ToastOptions = {
+    style: ToastStyle.Animated,
+    title: actionOptions.title,
+    message: actionOptions.message
+  };
+
+  let toast: Toast | undefined;
+  let controller: AbortController | undefined;
+
+  if (actionOptions.cancelable) {
+    controller = new AbortController();
+    options.primaryAction = {
+      title: "Cancel",
+      onAction: () => {
+        controller?.abort();
+        toast?.hide();
+      }
+    }
+  }
+
+  toast = new Toast(options);
+  toast.show();
```

Comment:
```suggestion
  let controller: AbortController | undefined;

  if (actionOptions.cancelable) {
    controller = new AbortController();
    options.primaryAction = {
      title: "Cancel",
      onAction: () => {
        controller?.abort();
        toast?.hide();
      }
    }
  }

  const toast = new Toast(options);
  toast.show();
```

Reply:
@mathieudutour I don't understand where does the `toast` variable come from on line 90?

But incredibly (and very surprisingly - to me) it works and the `toast` var is valid and non-null.

Reply:
variable hoisting in javascript 😬

Reply:
you could also write
```js
const toast = new Toast(options)

if (cancelable) {
  toast.primaryAction = ...
}

toast.show()
```